spring-pulsar/src/main/java/org/springframework/pulsar/core/CachingPulsarProducerFactory.java

```diff
@@ -205,13 +204,14 @@ public boolean equals(Object o) {
 			}
 			ProducerCacheKey<?> that = (ProducerCacheKey<?>) o;
 			return this.topic.equals(that.topic) && this.schemaHash.equals(that.schemaHash)
-					&& Objects.equals(this.router, that.router) && Objects.equals(this.interceptors, that.interceptors);
+					&& Objects.equals(this.encryptionKeys, that.encryptionKeys)
+					&& Objects.equals(this.customizers, that.customizers);
```

Comment:
I don't think customizers will be a reliable cache key. It is likely just a lambda w/ arbitrary code in it. Before we were using router+interceptors. Not sure what the answer is here but we need to dig on this one a bit.

Reply:
Before, a List of interceptors were used in the key. It doesn't seem better than using customizers.
Using a lambda is the same as with a class instance.
If you do
```java
List<ProducerBuilderCustomizer<String>> customizers1 = List.of(p -> p.compressionType(CompressionType.ZLIB));
producerFactory.createProducer(schema1, topic1, encryptionKeys1, customizers1);
producerFactory.createProducer(schema1, topic1, encryptionKeys1, customizers1);
```
the cache is used.
If you do
```java
producerFactory.createProducer(schema1, topic1, encryptionKeys1, List.of(p -> p.compressionType(CompressionType.ZLIB)));
producerFactory.createProducer(schema1, topic1, encryptionKeys1, List.of(p -> p.compressionType(CompressionType.ZLIB)));
```
the cache is not used.

I'll update the test to use a lambda instead of a mock as it's closer to real usage.

Reply:
I used lambdas in the cache key tests.